patches/kaggle_datasets.py

```diff
@@ -0,0 +1,24 @@
+import os
+from kaggle_web_client import KaggleWebClient
+
+_KAGGLE_TPU_NAME_ENV_VAR_NAME = 'TPU_NAME'
+
+class KaggleDatasets:
+    GET_GCS_PATH_ENDPOINT = '/requests/CopyDatasetVersionToKnownGcsBucketRequest'
+
+    # Integration types for GCS
+    AUTO_ML = 1
+    TPU = 2
+
+    def __init__(self):
+        self.web_client = KaggleWebClient()
+        self.has_tpu = os.getenv(_KAGGLE_TPU_NAME_ENV_VAR_NAME) is not None
+
+    def get_gcs_path(self, dataset_dir: str = None) -> str:
+        integration_type = self.TPU if self.has_tpu else self.AUTO_ML
```

Comment:
Wonder if explicitly printing which integration type you're using will help debug issues that may come up (ie. if the user meant to use one over the other).

Reply:
I actually had one for my own testing and debugging, but I think it would be a little strange for users in a notebook environment to have a utility method print output to the notebook. It would be nice if there was some kind of non-intrusive debug log for the notebooks. I'm not a big fan of the magic, but if we are setting TPU_NAME consistently then it seems like this should just work.